app/snippet.rb

```diff
@@ -0,0 +1,10 @@
+class Snippet
+  def self.generate(html)
+    html
+      .split('</h1>')[1..-1].join # make sure we skip anything before <h1>
+      .gsub(/(\<h.+\>)/, "") # remove headings
```

Comment:
It will also remove any other content in an HTML tag on the same line as the header.

Maybe it won't be a problem, but you can be sure by using a less greedy regex.

eg `/<h.>.+?<\/h.>/`